extensions/machine-learning/src/views/models/modelBrowsePage.ts

```diff
@@ -80,24 +80,24 @@ export class ModelBrowsePage extends ModelViewBase implements IPageView, IDataCo
 				if (this.localModelsComponent && this.azureModelsComponent && this.registeredModelsComponent) {
 					this.azureModelsComponent.removeComponents(this._formBuilder);
 					this.registeredModelsComponent.removeComponents(this._formBuilder);
-					this.localModelsComponent.addComponents(this._formBuilder);
 					await this.localModelsComponent.refresh();
+					this.localModelsComponent.addComponents(this._formBuilder);
 				}
 
 			} else if (this.modelSourceType === ModelSourceType.Azure) {
 				if (this.localModelsComponent && this.azureModelsComponent && this.registeredModelsComponent) {
 					this.localModelsComponent.removeComponents(this._formBuilder);
+					await this.azureModelsComponent.refresh();
```

Comment:
can you elaborate more on these ordering changes please?

Reply:
refresh loads the data  and add components creates the UI. so if the data is loaded after UI is loaded, we might show the UI for when there's no user